Content.Shared/CombatMode/SharedCombatModeSystem.cs

```diff
@@ -86,6 +85,7 @@ public virtual void SetInCombatMode(EntityUid entity, bool value, CombatModeComp
             return;
 
         SetMouseRotatorComponents(entity, value);
+        Dirty(entity, component);
     }
```

Comment:
This shouldn't be here, it will cause desyncs with combat mode because the state's not updating when combatmode actually changes.

Reply:
I see...

The error only goes away if I dirty the entity after MouseRotatorComps gets set or removed and I just wanted to avoid dirtying the entity twice by moving it down here

so should I solve this by moving the original dirty back for combat mode and add a DirtyEntity() after SetMouseRotatorComponents?

Reply:
I could also try using the EnsureComps to get a component variable to Dirty(entity, component) with

I don't know the beneficial differences between Dirty and DirtyEntity() at the moment, since the description of Dirty says that it implicitly dirties the entity the component belongs to...

Reply:
You should never be calling DirtyEntity as that dirties every single component and is needlessly expensive.

I haven't read the stacktrace so don't know the actual fix I just know this isn't it.

Reply:
> You should never be calling DirtyEntity as that dirties every single component and is needlessly expensive.
> 
> I haven't read the stacktrace so don't know the actual fix I just know this isn't it.

okay, then I'll move the dirtying specifically to the problematic comp in question and dirty it instead

Reply:
so despite the dirtying a comp summary saying that it implicitly dirties the entity, it's a different kind of dirty than DirtyEntity? even though if I follow the trail it uses DirtyEntity... I am slightly confused but uh, there we go